app/scripts/lib/validate.js

```diff
@@ -103,6 +103,47 @@ define(function (require, exports, module) {
       }
 
       return password.length >= Constants.PASSWORD_MIN_LENGTH;
+    },
+
+    /**
+     * Check whether data matches schema, depth-first.
+     * Compares using Object.prototype.toString.
+     */
+    isDataValid: function isDataValid (data, schema) {
+      if (! schema) {
+        return false;
+      }
+
+      if (typeof schema === 'string') {
+        if (schema[0] === '?') {
+          if (data === null || data === undefined) {
+            return true;
+          }
+
+          schema = schema.substr(1);
+        }
+
+        return Object.prototype.toString.call(data) === '[object ' + schema + ']';
```

Comment:
Imho no, but perhaps there is a benefit to doing so that I'm missing?

The way I see it, this function exists to validate dynamically-produced data, at runtime. The schemata are defined statically so we can validate them much earlier, when we run the tests. Validating them again on execution doesn't seem to buy us anything.